### YamlMime:ManagedReference
items:
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
    id: _provisdf54b54351d86c7bee42822d640658c9
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    children:
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException()
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException(String)
      - 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException(String,Throwable)'
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException(Throwable)
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.yml
    langs:
      - java
    name: ProvisioningServiceClientServiceException
    nameWithType: ProvisioningServiceClientServiceException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException
    type: Class
    source:
      remote: &ref_0
        path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientServiceException.java
        branch: master
        repo: 'https://github.com/Azure/azure-iot-sdk-java'
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientServiceException.java
      startLine: 10
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    summary: <p><span data-ttu-id="c1a99-101">これは、サービスによって報告された例外のデバイスのプロビジョニング サービス例外のサブセットです。</span><span class="sxs-lookup"><span data-stu-id="c1a99-101">This is the subset of the Device Provisioning Service exceptions for the exceptions reported by the Service.</span></span> </p>
    syntax:
      content: public class ProvisioningServiceClientServiceException extends ProvisioningServiceClientException
    inheritance:
      - java.lang.Object
      - Exception
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception
    inheritedMembers:
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException()
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException(String)
      - 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException(String,Throwable)'
      - com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException(Throwable)
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException()
    id: ProvisioningServiceClientServiceException()
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.yml
    langs:
      - java
    name: ProvisioningServiceClientServiceException()
    nameWithType: ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException()
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException()
    overload: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException*
    type: Constructor
    source:
      remote: *ref_0
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientServiceException.java
      startLine: 12
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    syntax:
      content: public ProvisioningServiceClientServiceException()
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException(String)
    id: ProvisioningServiceClientServiceException(String)
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.yml
    langs:
      - java
    name: ProvisioningServiceClientServiceException(String message)
    nameWithType: ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException(String message)
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException(String message)
    overload: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException*
    type: Constructor
    source:
      remote: *ref_0
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientServiceException.java
      startLine: 17
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    syntax:
      content: public ProvisioningServiceClientServiceException(String message)
      parameters:
        - id: message
          type: '26831127'
  - uid: 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException(String,Throwable)'
    id: 'ProvisioningServiceClientServiceException(String,Throwable)'
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.yml
    langs:
      - java
    name: 'ProvisioningServiceClientServiceException(String message, Throwable cause)'
    nameWithType: 'ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException(String message, Throwable cause)'
    fullName: 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException(String message, Throwable cause)'
    overload: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException*
    type: Constructor
    source:
      remote: *ref_0
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientServiceException.java
      startLine: 22
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    syntax:
      content: 'public ProvisioningServiceClientServiceException(String message, Throwable cause)'
      parameters:
        - id: message
          type: '26831127'
        - id: cause
          type: a04fafd7
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException(Throwable)
    id: ProvisioningServiceClientServiceException(Throwable)
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.yml
    langs:
      - java
    name: ProvisioningServiceClientServiceException(Throwable cause)
    nameWithType: ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException(Throwable cause)
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException(Throwable cause)
    overload: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException*
    type: Constructor
    source:
      remote: *ref_0
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientServiceException.java
      startLine: 27
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    syntax:
      content: public ProvisioningServiceClientServiceException(Throwable cause)
      parameters:
        - id: cause
          type: a04fafd7
references:
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.ProvisioningServiceClientServiceException*
    name: ProvisioningServiceClientServiceException
    nameWithType: ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException.ProvisioningServiceClientServiceException
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
  - uid: '26831127'
    spec.java:
      - name: String
        fullName: String
  - uid: a04fafd7
    spec.java:
      - name: Throwable
        fullName: Throwable
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException()
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.yml
    name: ProvisioningServiceClientException()
    nameWithType: ProvisioningServiceClientException.ProvisioningServiceClientException()
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientException.ProvisioningServiceClientException()
    type: Constructor
    syntax:
      content: public ProvisioningServiceClientException()
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException(String)
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.yml
    name: ProvisioningServiceClientException(String message)
    nameWithType: ProvisioningServiceClientException.ProvisioningServiceClientException(String message)
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientException.ProvisioningServiceClientException(String message)
    type: Constructor
    syntax:
      content: public ProvisioningServiceClientException(String message)
      parameters:
        - id: message
          type: '26831127'
  - uid: 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException(String,Throwable)'
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.yml
    name: 'ProvisioningServiceClientException(String message, Throwable cause)'
    nameWithType: 'ProvisioningServiceClientException.ProvisioningServiceClientException(String message, Throwable cause)'
    fullName: 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientException.ProvisioningServiceClientException(String message, Throwable cause)'
    type: Constructor
    syntax:
      content: 'public ProvisioningServiceClientException(String message, Throwable cause)'
      parameters:
        - id: message
          type: '26831127'
        - id: cause
          type: a04fafd7
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.ProvisioningServiceClientException(Throwable)
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.yml
    name: ProvisioningServiceClientException(Throwable cause)
    nameWithType: ProvisioningServiceClientException.ProvisioningServiceClientException(Throwable cause)
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientException.ProvisioningServiceClientException(Throwable cause)
    type: Constructor
    syntax:
      content: public ProvisioningServiceClientException(Throwable cause)
      parameters:
        - id: cause
          type: a04fafd7
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisioning_service_client_exception.yml
    name: ProvisioningServiceClientException
    nameWithType: ProvisioningServiceClientException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientException
    type: Class
    summary: |-
      <p><span data-ttu-id="c1a99-102">スーパー サービス クライアントでデバイスのプロビジョニング サービス例外クラスです。</span><span class="sxs-lookup"><span data-stu-id="c1a99-102">Super class for the Device Provisioning Service exceptions on the Service Client.</span></span></p>
      <p>

        <pre>

          <code>ProvisioningServiceClientException

              |

              +--&gt;ProvisioningServiceClientTransportException [any transport layer exception]

              |

              +--&gt;ProvisioningServiceClientServiceException [any exception reported in the http response]

                      |

                      |

                      +--&gt;ProvisioningServiceClientBadUsageException [any http response 4xx]

                      |        |

                      |        +--&gt;ProvisioningServiceClientBadFormatException [400]

                      |        +--&gt;ProvisioningServiceClientUnathorizedException [401]

                      |        +--&gt;ProvisioningServiceClientNotFoundException [404]

                      |        +--&gt;ProvisioningServiceClientPreconditionFailedException [412]

                      |        +--&gt;ProvisioningServiceClientTooManyRequestsException [429]

                      |

                      +--&gt;ProvisioningServiceClientTransientException [any http response 5xx]

                      |        |

                      |        +--&gt;ProvisioningServiceClientInternalServerErrorException [500]

                      |

                      +--&gt;ProvisioningServiceClientUnknownException [any other http response &gt;300, but not 4xx or 5xx]

          </code>

        </pre>

      </p>
    syntax:
      content: public class ProvisioningServiceClientException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8d640264fa5997294ef7a61f3c9f7e67
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8d640264fa5997294ef7a61f3c9f7e67.yml
    name: ProvisioningServiceClientBadUsageException
    nameWithType: ProvisioningServiceClientBadUsageException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientBadUsageException
    type: Class
    summary: |-
      <p><span data-ttu-id="c1a99-103">これは、使用法エラーに関連する例外のデバイスのプロビジョニング サービス例外のサブセットです。</span><span class="sxs-lookup"><span data-stu-id="c1a99-103">This is the subset of the Device Provisioning Service exceptions for the exceptions related to usage error.</span></span></p>
      <p><span data-ttu-id="c1a99-104">プロビジョニング サービスが応答不適切な使用法は、例外のいずれかの http 要求、rest API が正しく呼び出された場合、使用状況の問題には、インスタンスのユーザーは、その操作の権限がありません。</span><span class="sxs-lookup"><span data-stu-id="c1a99-104">The provisioning service will response a http request with one of the bad usage exception if the rest API was properly called, with a usage issue, for instance the user is not authorized for that operation.</span></span> </p>
      <p><span data-ttu-id="c1a99-105">HTTP ステータス コード 400 ~ 499 です。</span><span class="sxs-lookup"><span data-stu-id="c1a99-105">HTTP status code 400 to 499.</span></span> </p>
    syntax:
      content: public class ProvisioningServiceClientBadUsageException extends ProvisioningServiceClientServiceException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a0fa33fb6d65f102e28e9bcde677ce1
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a0fa33fb6d65f102e28e9bcde677ce1.yml
    name: ProvisioningServiceClientTransientException
    nameWithType: ProvisioningServiceClientTransientException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientTransientException
    type: Class
    summary: |-
      <p><span data-ttu-id="c1a99-106">これは、例外が、一時的なサービスの問題を関連するデバイスのプロビジョニング サービス例外のサブセットです。</span><span class="sxs-lookup"><span data-stu-id="c1a99-106">This is the subset of the Device Provisioning Service exceptions for the exceptions related a temporary service issue.</span></span></p>
      <p><span data-ttu-id="c1a99-107">プロビジョニング サービスは、rest API は、適切に呼び出されが、サービスがその時点でそのアクションを実行することではない場合に、一時的な例外のいずれかの http 要求になるように応答します。</span><span class="sxs-lookup"><span data-stu-id="c1a99-107">The provisioning service will response a http request with one of the transient exception if the rest API was properly called, but the service is not able to execute that action at that time.</span></span> <span data-ttu-id="c1a99-108">これらは、再試行に役立つ、問題を解決する例外です。</span><span class="sxs-lookup"><span data-stu-id="c1a99-108">These are the exceptions that a retry can help to fix the issue.</span></span> </p>
      <p><span data-ttu-id="c1a99-109">HTTP 状態コード 500 を 599 です。</span><span class="sxs-lookup"><span data-stu-id="c1a99-109">HTTP status code 500 to 599.</span></span> </p>
    syntax:
      content: public class ProvisioningServiceClientTransientException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0b465facdcea6be4aadae8932ae4d42c
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0b465facdcea6be4aadae8932ae4d42c.yml
    name: ProvisioningServiceClientUnknownException
    nameWithType: ProvisioningServiceClientUnknownException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientUnknownException
    type: Class
    summary: |-
      <p><span data-ttu-id="c1a99-110">これは、不明な問題のデバイスのプロビジョニング サービス例外のサブセットです。</span><span class="sxs-lookup"><span data-stu-id="c1a99-110">This is the subset of the Device Provisioning Service exceptions for the unknown issues.</span></span></p>
      <p><span data-ttu-id="c1a99-111">HTTP ステータス コード 300 + がない 4nn または 5nn します。</span><span class="sxs-lookup"><span data-stu-id="c1a99-111">HTTP status code 300+, but not 4nn or 5nn.</span></span> </p>
    syntax:
      content: public class ProvisioningServiceClientUnknownException